### DISCUSSION TITLE
throw render() with i18n with client routing doesn't work properly

### DISCUSSION THREAD

**Discussion by lebretont** [vikejs/vike#1436] [Discussion] [answered]
Hi,

I'm currently learning Vike, congratulations for this amazing project.

I want to render a certain page based on a guard. 
When the locale is not the default one, the page is not translated on client routing : 
![oMSMlBg](https://github.com/vikejs/vike/assets/59013432/691f7176-cf7b-455e-bb3d-399cb905c1af)

After a little bit of investigation, adding "+onBeforeRender" makes the client fetching data on "xxx/index.pageContext.json" on each route change.

But i haven't found a method to get this page translated.

I tried this `throw render('/' + pageContext.locale + '/about')` on my guard but i get 
> URL /fr-FR/about doesn't match the route of any of your pages

Is there something I'm not doing right?

Here is a minimal reproduction : https://github.com/lebretont/vike-i18n-client.

Thanks for the help!

**Comment by brillout** [comment]
Indeed, I could reproduce the issue. I'll have a closer look at it tomorrow.

**Reply by brillout** [reply]
I'll resume tomorrow. @lebretont I guess this is a blocker and therefore urgent for you, correct?

**Comment by lebretont** [comment]
I'm still playing with vike so this is not urgent.
Thanks for reviewing it.

**Reply by brillout** [reply]
Ok, let me first finish writing the docs for `vike-{react,vue,solid}` then I'll fix this. FYI I consider this a Vike bug so it's a top priority. ETA ~~this week~~ today.

**Reply by brillout** [reply]
Improved reproduction: https://github.com/brillout/vike-i18n-throw-render.

**Reply by brillout** [reply, marked as answer]
[Fix](https://github.com/vikejs/vike/commit/d8d55d27b018eddb55d1c7205d7fcb366bf4a1e5) pre-released in `0.4.156-commit-d8d55d2`.

Also make sure to [correctly use `throw render()`](https://github.com/brillout/vike-i18n-throw-render/commit/bb93ed817aaa52873090fa83a64e0d136d7b56ce) and [use `pageContext.urlPathname` instead of `pageContext.urlOriginal`](https://github.com/vikejs/vike/commit/0a6b196bce65f67bec717e775f120c5765abff0d). (Further improving docs about this is on the radar.)

Apologies for the delayed handling of this bug. It wasn't a trivial one and writing docs for `vike-{react,vue,solid}` is getting quite important. Usually such bugs are fixed under 24h.

I guess it's too early but would your company, eventually, be up for [sponsoring](https://github.com/vikejs/vike/issues/1350)?

**Reply by lebretont** [reply]
Thanks for you fast fix!
Unfortunately my company doesn't use Vike. But I sponsored as an individual.

I already made my portfolio and currently working on a more complete website using many features and it works like a charm :)
I really want to thank you and congratulate you on this amazing project. Vike is so simple and customizable, it makes frontend development very enjoyable.

**Reply by lourot** [reply]
Thank you for sponsoring @lebretont :pray:

**Reply by brillout** [reply]
Thank you for the donation, and I'm glad you're enjoying Vike.
